README.md

```diff
@@ -1,32 +1,81 @@
 # oci-arch-custom-dns-fn
 
-[![License: UPL](https://img.shields.io/badge/license-UPL-green)](https://img.shields.io/badge/license-UPL-green) [![Quality gate](https://sonarcloud.io/api/project_badges/quality_gate?project=oracle-devrel_oci-arch-custom-dns-fn)](https://sonarcloud.io/dashboard?id=oracle-devrel_oci-arch-custom-dns-fn)
-
-## THIS IS A NEW, BLANK REPO THAT IS NOT READY FOR USE YET.  PLEASE CHECK BACK SOON!
+[![License: UPL](https://img.shields.io/badge/license-UPL-green)](https://img.shields.io/badge/license-UPL-green)  [![Quality gate](https://sonarcloud.io/api/project_badges/quality_gate?project=oracle-devrel_oci-arch-custom-dns-fn)](https://sonarcloud.io/dashboard?id=oracle-devrel_oci-arch-custom-dns-fn) 
 
 ## Introduction
-MISSING
+
+The DNS Private Zone feature within OCI's DNS service allows the users to expand the capabilities of the built-in VCN DNS resolver, enabling them to use custom domain names for the resources within the VCNs.
+
+This sample deployment uses OCI Events to monitor a parent compartment for compute resource creation and termination. When such an event occurs, OCI Functions are triggered to maintain the records in a user-configured DNS Zone. 
+
+In case of duplicate records, the function will attempt to generate a new hostname using the latest five characters in instance OCID.
+
+Two free-form tags: `hostnames` and `dns_zone_record_hashes`, will be attached to the instance to confirm the DNS Zone record insertion at instance creation and DNS Zone record deletion at instance termination.
```

Comment:
we started to adopt a pattern of also adding free form tags naming the module/ repo that was used as the source.  Helped back track if there was a need to help a customer. Plus if a TF template was updated the customer coult determine if they have deployed it